dds/idl/dds_generator.cpp

```diff
@@ -14,6 +14,45 @@ using namespace AstTypeClassification;
 
 dds_generator::~dds_generator() {}
 
+bool dds_generator::cxx_escape(const std::string& s, size_t i)
+{
+  const std::string cxx = "_cxx_";
+  return s.substr(i, cxx.size()) == cxx;
+}
+
+std::string dds_generator::valid_var_name(const std::string& str)
+{
+  // Replace invalid characters with a single underscore
+  const std::string invalid_chars("<>()[]*.: ");
+  std::string s;
+  char last_char = '\0';
+  for (size_t i = 0; i < str.size(); ++i) {
+    const char c = str[i];
+    if (invalid_chars.find(c) == std::string::npos) {
+      s.push_back(c);
+      last_char = c;
+      if (c == '_') {
+        s.push_back('9');
+        last_char = '9';
+      }
+    } else if (last_char != '_') {
+      s.push_back('_');
+      last_char = '_';
+    }
+  }
```

Comment:
```suggestion
  }

```

This part is doing something separate from what the comment is saying and originally had its own comment to say what it was doing.

Reply:
Can you elaborate what you mean by this?

Reply:
This is still the same.

Reply:
What exactly are you referring to?